### PR TITLE
Feature/change pie chart to dashboard

### PR DESCRIPTION
**Pull request by jmcarcedo**
### Checklist

- [x] I've validated if any R8 rule should be added to the relevant proguard-rules.pro file. More info about it [on Confluence](https://futureworkshops.atlassian.net/wiki/spaces/FMW/pages/2319187983/Validate+R8+rules+for+core+and+plugins)
- [x] If I'm updating a plugin submodule, I'm sure that the reference on the submodule is on `main`

### Task

[Charts / Dashboard step](https://3.basecamp.com/5245563/buckets/26145695/todos/4870042132)

### Feature/Issue

Update charts submodule to V4 replacing the deprecated steps `Pie Chart` and `Network Pie Chart` for the new `Dashboard Chart` and `Network Dashboard Chart`

### Implementation

* Create new data models for both steps.
* Refactor steps and plugins to adapt to the new models
* Create new view and adapter to follow design
* Create new result model adapted to Charts
* Implements new color palette for both light and dark modes

### CODE REVIEW

File: charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/ItemProvider.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            provider: (List<DashboardBaseChartItem>) -> Unit,
+            onLoading: () -> Unit,
+            onError: () -> Unit
+        ) {
+            val formalisedItems = items.map {
+                DashboardBaseChartItem(
+                    id = it.id,
+                    listItemId = it.listItemId,
+                    chartType = it.chartType,
+                    title = it.title,
```
Comment (igorcferreira) [inline charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/ItemProvider.kt]: @jmcarcedo It is important to pass these strings through the localization services. Especially title and subtitle. To ensure that (even more for the static dashboard) these strings can be translated using the App Builder. The comment is also valid for the network response (from the load method) down below.

File: charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/UINetworkChartPluginStep.kt [hunk 2]
```diff
@@ -1,49 +1,56 @@
...
+
     override fun createView(
         stepResult: AnswerResult?,
         services: ServiceBox,
         appServiceResponse: AppServiceResponse
     ): FragmentStep {
-        val fullUrl = "${appServiceResponse.server?.url}/${url}"
-        
-        val task = URLIAsyncTask<Nothing, List<PieChartItem>>(
+        val fullUrl = "${appServiceResponse.server?.url}${url}"
```
Comment (igorcferreira) [inline charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/UINetworkChartPluginStep.kt]: @jmcarcedo URLs should be resolved using URL resolver services to be sure to have a valid session values translation. Like, for example, `/chart/{uuid.answer}` being resolved from the session.

**Review by igorcferreira — CHANGES_REQUESTED** [review]

File: charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/ItemProvider.kt (hunk 1 above)
Comment (jmcarcedo) [reply charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/ItemProvider.kt]: Changed in [0179af8](https://github.com/FutureWorkshops/MWChartsPlugin-Android/pull/12/commits/0179af8b501acb4626143b73badd1e5a2c7765a9)

File: charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/UINetworkChartPluginStep.kt (hunk 2 above)
Comment (jmcarcedo) [reply charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/UINetworkChartPluginStep.kt]: Changed in [787e972](https://github.com/FutureWorkshops/MWChartsPlugin-Android/pull/12/commits/787e9721ac6c7bafc36978b5e9fa1ddf707ab3b5)

**Review by igorcferreira — APPROVED** [review]

**Pull request merged**